### DISCUSSION TITLE
What is the relationship between openmp compilation options and C.MemoryResource configuration?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6932] [Development]
### ImageMagick version

7.1.0-52

### Operating system, version and so on

Debian 6.3.0-18+deb9u1

### Description

If **openmp** is turned off when compiling, no matter how much **MagickSetResourceLimit** is set to **C.MemoryResource**, a single thread will be used?

**Comment by urban-warrior** [comment]
When OpenMP is disabled, all processes operate as single-threaded entities. However, by enabling OpenMP, certain image processing algorithms can experience accelerated performance through parallel processing of multiple rows in your image simultaneously. It's important to be mindful that in some instances, OpenMP may necessitate additional memory compared to its single-threaded counterpart, as local memory per thread needs to be established.

The memory resource limit plays a crucial role in governing the amount of memory an image cache can utilize. Should this limit be surpassed, the pixels are alternatively cached to disk. Notably, this resource limit exclusively impacts the pixel cache, and it's essential to recognize that ImageMagick may consume memory to support the image-processing algorithm independently of the memory resource limit.
